entrypoints/content/components/secondary/comment/CommentAction.tsx

```diff
@@ -0,0 +1,71 @@
+
+import { useState } from 'react'
+
+
+// Components::
+import { ArrowBigDown, ArrowBigUp, Ellipsis, Forward, MessageSquare } from 'lucide-react'
+
+// Types:
+import { VoteCount } from 'types/votes'
+
+interface VoteCountProps {
+  voteCount: VoteCount
+}
+
+const CommentAction: React.FC<VoteCountProps> = ({ voteCount: { up = 0, down = 0 } }) => {
+
+  const [voteCount, setVotecount] = useState(up - down)
+  const [isUpVoteActive, setUpVoteActive] = useState(true)
+  const [isDownVoteActive, setDownVoteActive] = useState(false)
+
+  useEffect(() => {
+    setVotecount(up - down)
+  }, [up, down])
+
+  // #TODO: Logic is incorrect need to fix this
+  // Maybe take all the states into one single object state
+  const upVote = () => {
+    if (isUpVoteActive) {
+      setUpVoteActive(false)
+      setVotecount(voteCount - 1)
+      return
+    }
+    setUpVoteActive(true)
+    setDownVoteActive(false)
+    setVotecount(voteCount + 1)
+  }
+  const downVote =  () => {
+    if (isDownVoteActive) {
+      setDownVoteActive(false)
+      setVotecount(voteCount + 1)
+      return
+    }
+    setUpVoteActive(false)
+    setDownVoteActive(true)
+    setVotecount(voteCount - 1)
+  }
+
+  return (
+    <div className='flex space-x-6 items-center pt-1 -ml-[px]'>
+      <div className='flex space-x-2 items-center'>
+        <ArrowBigUp  size={18} {...(isUpVoteActive && { fill: '#059669'  }) }  color={ isUpVoteActive ? '#059669' : 'currentColor'} onClick={upVote} />
```

Comment:
we can use tailwind here instead of using conditional props

also, we're hardcoding the hex value here which will bite when the application grows

simply, `className={cn(isUpvoteActive ? 'text-green fill-green' : 'text-brand-primary fill-none', 'transition-all')}`
^this makes use of the tailwindcss custom brand colors, doesn't hardcode anything, and also introduces transitions

Reply:
also, add `cursor-pointer` to the class list to make the cursor go pointy when hovering over the up and down icons

Reply:
Lucide-react icons does not work like that have to figure out some other way to pass it down to props properly